pkg/server/image_remove.go

```diff
@@ -51,15 +51,12 @@ func (c *criContainerdService) RemoveImage(ctx context.Context, r *runtime.Remov
 			}
 			return nil, fmt.Errorf("failed to get image %q: %v", tag, err)
 		}
-		desc, err := cImage.Config(ctx)
```

Comment:
I think we should use image id here. Why change it to manifest digest?

Reply:
We can get the manifest digest from the struct directly.  but we only get image id by reading Config from content. If the content of config is missing, we will fail to get image id,we will can't remove the image forever.

Reply:
That's a valid concern.

Did you see that in your environment?

Reply:
Logically we should still use image id. If you are concern with corrupted image, can we check not found error instead?

Reply:
I saw a problem. But it is ok after we delete dir "/var/lib/containerd":)
So I don't know the reason. just guess it.

The Config() function will read minifest, config from content. It is difficult to check that it is not found.

Reply:
The problem is that using manifest digest here is wrong.
Our image is indexed by image ID. Removing an image means removing all references associated with this image id.
It's possible that 2 manifest digests have the same image ID in theory.

Reply:
@yanxuean WDYT? If you decide to do this, please add comment in your code to explain we should do that. :)

Reply:
>It's possible that 2 manifest digests have the same image ID in theory.

Is this from upgraded image. For example, I create a new image by only adding a layer. the new image  will has same config with old image, but their minifest is different.

Reply:
If 2 manifest digests have the same image ID . There is a new problem.
For example. 
1. image1:    minifest1, imageID1(from config), repoTag1, repoTag2
2. image2:   minifest2, imageID1(from config),  repoTag1, repoTag3
We pull image1 first, then image2.  The repoTag1 in imageService refers to minifest2 now.
We want to remove image1, Our current RemoveImage() will successfully remove the repoTag1 from imageService. because the imageID1 is unmodified.

Reply:
@yanxuean `RootFS` is uncompressed. Manifest digest is generated from compressed content.

In theory it's possible that an image is compressed with different algorithm, then they'll have the same uncompressed id - image id, but different ids generated from compressed contents -
 manifest digest.

Reply:
>@yanxuean RootFS is generated from uncompressed layers. Manifest digest is generated from compressed content.

>In theory it's possible that an image is compressed with different algorithms, then they'll have the same uncompressed id - image id, but different ids generated from compressed contents - manifest digest.

@Random-Liu Manifest digest seems to be generated from `manifest`, but it's a json file.
Did you mean that `manifest` contain the info of compressed layers(e.g, `mediaType`, `size`) and layers may be compressed with different algorithms?